js/ui/status/keyboard.js

```diff
@@ -480,6 +481,40 @@ const InputSourceManager = new Lang.Class({
             this._updateMruSettings();
     },
 
+    passwordMode: function(enable) {
+        let source;
+
+        if (enable) {
+            // do not switch the keyboard layout if the current one is a latin one
+            if (this._keyboardManager.isLatinLayout(this._currentSource.id))
+                return;
+
+            // look for latin keyboard layout to switch to
+            for (let i = 0; i < this._mruSources.length; ++i) {
+                let src = this._mruSources[i];
+                if (this._keyboardManager.isLatinLayout(src.id)) {
+                    source = src;
+                    break;
+                }
+            }
+            if (!source)
+                // fallback to an us keyboard layout
+                source = new InputSource('xkb',
+                                         'us',
+                                         'English (US)',
+                                         'us',
+                                         0);
+            this._lastSource = this._currentSource;
+        } else {
+            if (!this._lastSource)
+                return;
+            source = this._lastSource;
+            this._lastSource = null;
```

Comment:
Do we need to remove the fallback `us` here source if we added it before?

Reply:
Yes, actually that code was a bit uncomplete, I had to add it to the list of layouts as well and signal that change in order to be picked up.